### ISSUE TITLE
[MU4 Issue] MuseScore Failing to Open on Windows 11

### ISSUE THREAD

**Issue by DaffodilAura** [musescore/MuseScore#12577]
MuseScore 4 closes or crashes at start.

**To Reproduce**
1. Open MuseScore 4
2. Wait until the Getting Started window appears
3. See crash

**Expected behavior**
MuseScore should not crash or close at start.

**Platform information**
 - OS: Windows 11

**Additional context**
This is my first time attempting to download and run a nightly build.

**Comment by DaffodilAura** [comment]
https://user-images.githubusercontent.com/110142307/181415109-1ab5e522-7a34-4549-bb35-04509dfebdc2.mp4

I need to share this video, which shows the issue in action.

**Comment by Eism** [comment]
@DaffodilAura 
I can't reproduce on the latest nightly build
Please, attach a log file from C:\Users\<user>\AppData\Local\MuseScore\MuseScore4Development\logs

**Comment by DaffodilAura** [comment]
What is a log file?

**Comment by DaffodilAura** [comment]
I found what was causing it...  The firewall was blocking the app from communicating with the computer.

**Comment by HemantAntony** [comment]
Norton keeps on telling me that the connection between MuseScore and the Internet isn't secure. Well, at least it was saying that a few days ago

**Comment by cbjeukendrup** [comment]
Some background about that firewall problem: I believe it's caused by Qt's implementation of the OAuth flow. For the OAuth flow, you need to specify a redirect URL, to which the user will be redirected when the authorization process is complete. It looks like MuseScore starts acting as a server and opens some port, so that the redirect URL for the OAuth flow can be a `http://localhost:…` URL. This is why a lot of firewalls complain about MuseScore 4.

Other apps use a different implementation: they register the app with the OS for a special URL scheme, for example `x-musescore-auth://`. The redirect URL is in this case a URL with that special scheme. When the OS is asked to handle this URL, it will know to open MuseScore. Your browser may ask something like "Do you want to allow musescore.com to open MuseScore?" 
This implementation is used by pretty much all OAuth-using apps I know, including GitHub Desktop and Microsoft Office. The advantage is that it is safe and doesn't get blocked by firewalls etc. The same implementation can be used for example to open a file from a web service in a desktop app, like you can do with OneDrive for example. 
The disadvantage is that it strongly differs per OS, and is probably not available on most Linux distros, and that it requires special configuration for each OS, which can't be done simply in C++ code. So, it's absolutely not portable / cross-platform. That must be the reason that Qt came up with their (in my view unusual) implementation.

If MuseScore's firewall problem appears to cause people serious trouble, we could turn this into a separate issue.

**Comment by RobFog** [comment]
> What is a log file?

[Wikipedia knows.](https://en.wikipedia.org/wiki/Logging_(software))

In this case, you'd find it in the directory Eism mentioned: C:\Users<user>\AppData\Local\MuseScore\MuseScore4Development\logs

**Comment by DaffodilAura** [comment]
Makes sense to me @cbjeukendrup...
